DQMServices/Demo/test/test_dqmgui_upload.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/env bash
+# Simple test to use one of the files generated by TestDQMServicesDemo test
+# to upload it to the dev DQMGUI, making sure that the ROOT file format generated
+# is compatible. See cmssw issue #43590.
+# Requires curl and jq to be installed. Also requires either a Grid key and certificate,
+# or a proxy to be available.
+set -x
+set -e
+
+DEV_DQMGUI_URL="https://cmsweb.cern.ch/dqm/dev"
+# Create a unique fake "Era", so that different executions of this test
+# produce differently named files. This is required, because this test might pass 
+# if it's checking a file that was successfully uploaded during a previous
+# instance of the same test.
+UNIQUE_ERA_ID=$(date -u +%Y%m%d%M%S%N)
+OLD_FILENAME=DQM_V0001_R000000001__Harvesting__DQMTests__DQMIO.root
+NEW_FILENAME=$(echo $OLD_FILENAME | sed "s/DQMTests/DQMTests${UNIQUE_ERA_ID}/")
+# File size, to verify later that the file is fully uploaded
+original_file_size=$(stat --printf="%s" $OLD_FILENAME)
+
+# TODO: Is there any way to not hardcode the required test name, TestDQMServicesDemo?
+UNIT_TESTS_BASE_PATH=${LOCALTOP}/unit_tests/TestDQMServicesDemo
+
+# Make sure the required file exists. We expect to find it in the current directory.
+if [ ! -f $OLD_FILENAME ]; then
+    echo "Unable to find required file ${OLD_FILENAME}!"
+    exit 1
+fi
+
+# Prepare curl args
+curl_args=""
+if [ -n "$X509_CERT_DIR" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_PROXY} --cert ${X509_USER_PROXY}"
+elif [ -n "$X509_USER_CERT" ] && [ -n "$X509_USER_CERT" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_KEY} --cert ${X509_USER_CERT}"
+elif [ -f $HOME/.globus/usercert.pem ] && [ -f $HOME/.globus/userkey.pem ]; then
+    curl_args="-k --key $HOME/.globus/userkey.pem --cert $HOME/.globus/usercert.pem"
+else
+    echo "Unable to find proxy or key/cert env vars, cannot continue"
+    exit 1
+fi
+
+# Make a copy with a filename that conforms to the naming convention required by DQMGUI.
+rm -f $NEW_FILENAME
+cp $OLD_FILENAME $NEW_FILENAME
+
+# Upload the file to the DQMGUI
+visDQMUpload.py "$DEV_DQMGUI_URL" $NEW_FILENAME
+
+# Wait for a bit for the file to be imported into the DQMGUI's index.
+SECONDS_TO_WAIT=60
+# Reset shell's internal SECONDS var.
+SECONDS=0
+
+while true; do
+    # Wait for the dataset to appear in the /json/samples endpoint. It might take a while.
+    if curl -L $curl_args "${DEV_DQMGUI_URL}/data/json/samples?match=DQMTests${UNIQUE_ERA_ID}" | jq .samples[0].items[0].dataset | grep -q $UNIQUE_ERA_ID; then
+	exit 0
+    fi
+
+    sleep 1
+    # Timeout after SECONDS_TO_WAIT have passed.
```

Comment:
@nothingface0 , looks like we are not incrementing `SECONDS` here ... so in case of `curl ` failure we will not get out this loop

Reply:
This is [internal to the shell](https://man7.org/linux/man-pages/man1/bash.1.html) AFAIK:

>   **SECONDS**
              Each time this parameter is referenced, it expands to the
              number of seconds since shell invocation.  If a value is
              assigned to SECONDS, the value returned upon subsequent
              references is the number of seconds since the assignment
              plus the value assigned.  The number of seconds at shell
              invocation and the current time are always determined by
              querying the system clock.  If SECONDS is unset, it loses
              its special properties, even if it is subsequently reset.

I can see it incrementing in my tests, due to `set -x`.

Reply:
ah ok, good to know